### ISSUE TITLE
AutoMLPipelineBuilder.get_many_models_train_steps - Error "Environment name can not start with the prefix AzureML..."

### ISSUE THREAD

**Issue by crisansou** [Azure/MachineLearningNotebooks#1468]
Hello,

When running the experiment, the error message **Environment name can not start with the prefix AzureML** was displayed. How can I set the name of the environment? I'm following the GitHub tutorial and haven't found anything about it.

![image](https://user-images.githubusercontent.com/29444086/117882725-01767d80-b281-11eb-8df5-36d8683523e7.png)

Code used:

- Registering Dataset

![image](https://user-images.githubusercontent.com/29444086/117883192-81044c80-b281-11eb-9dec-d73431948061.png)

![image](https://user-images.githubusercontent.com/29444086/117883230-8c577800-b281-11eb-8445-060839369fe5.png)

- Training Pipeline

![image](https://user-images.githubusercontent.com/29444086/117883313-a5602900-b281-11eb-818d-3972111d7f9c.png)

![image](https://user-images.githubusercontent.com/29444086/117883356-b315ae80-b281-11eb-99d9-1ac6c0989186.png)

![image](https://user-images.githubusercontent.com/29444086/117883429-c7f24200-b281-11eb-88de-3979570adb55.png)

![image](https://user-images.githubusercontent.com/29444086/117883465-d2144080-b281-11eb-8b9c-f74756bedd01.png)

![image](https://user-images.githubusercontent.com/29444086/117883535-e48e7a00-b281-11eb-9d31-e035f44a0871.png)

References:

- https://github.com/microsoft/solution-accelerator-many-models/tree/master/Automated_ML/02_AutoML_Training_Pipeline
- https://github.com/MicrosoftDocs/azure-docs/issues/65770

Best regards,
Cristina

---
#### Detalhes do documento

⚠ *Não edite esta seção. É necessário para a vinculação do problema do docs.microsoft.com ➟ GitHub.*

* ID: 49399a7d-d4e8-370e-ce62-d60a6b64e412
* Version Independent ID: 782d8ba4-75dd-27c3-5a46-a921c3ead4bf
* Content: [azureml.contrib.automl.pipeline.steps.AutoMLPipelineBuilder class - Azure Machine Learning Python](https://docs.microsoft.com/pt-br/python/api/azureml-contrib-automl-pipeline-steps/azureml.contrib.automl.pipeline.steps.automlpipelinebuilder?view=azure-ml-py)
* Content Source: [AzureML-Docset/stable/docs-ref-autogen/azureml-contrib-automl-pipeline-steps/azureml.contrib.automl.pipeline.steps.AutoMLPipelineBuilder.yml](https://github.com/MicrosoftDocs/MachineLearning-Python-pr.pt-BR/blob/live/AzureML-Docset/stable/docs-ref-autogen/azureml-contrib-automl-pipeline-steps/azureml.contrib.automl.pipeline.steps.AutoMLPipelineBuilder.yml)
* Service: **machine-learning**
* Sub-service: **core**
* GitHub Login: @DebFro
* Microsoft Alias: **debfro**

**Comment by crisansou** [comment]
Hi,

I have some updates:

- I put the code below to set the environment.

from azureml.core.environment import Environment

env = Environment.get(workspace=ws, name="AzureML-Tutorial")
myenv = env.clone("automl_env")

train_steps = AutoMLPipelineBuilder.get_many_models_train_steps(experiment=experiment,
                                                                automl_settings=automl_settings,
                                                                train_data=dataset_input,
                                                                compute_target=compute_target,
                                                                partition_column_names=partition_column_names,
                                                                node_count=1,
                                                                process_count_per_node=2,
                                                                run_invocation_timeout=3700,
                                                                train_env=myenv)

- The environment problem has been resolved, but now the process displays the message **ValueError: None is not in list**. I don't know what this means.

![image](https://user-images.githubusercontent.com/29444086/118014360-82894f80-b329-11eb-8e6a-558d6606d7b1.png)

Best regards,
Cristina

**Comment by shbijlan** [comment]
@crisansou is there any error surfaced in 70_driver_log?

**Comment by crisansou** [comment]
Hi @shbijlan ,

I deleted the workspace. I tried to reproduce the steps again but I couldn't even create the experiment, below is the error message. Can you tell which is the recommended version to use this solution?

![image](https://user-images.githubusercontent.com/29444086/119876912-c752e000-befe-11eb-9a18-c8f03afae48c.png)

I don't know if it's related, but I realized that now compute instance is using version 1.29.

!pip install --upgrade azureml-sdk[automl]
!pip install azureml-contrib-automl-pipeline-steps

![image](https://user-images.githubusercontent.com/29444086/119877097-03864080-beff-11eb-8134-07d22a243284.png)




> @crisansou is there any error surfaced in 70_driver_log?

**Comment by crisansou** [comment]
Hi @deeptim123 ,

Thanks for the instructions! After including the environment I was able to run the cell, but the pipeline ended with an error because I am using a regression model.

In the next release, in addition to fixing the bug, will it be possible to use regression?

> from azureml.core. import Environment
> train_env = Environment.get(workspace = ws, name = 'AzureML-AutoML')
> 
> Can you please pass train_env like above for the workaround? There is a bug in our code that needs to be fixed. we will fix it in our next release.
> 
> Also this solution only supports 'forecasting' and needs a time_column_name passed in automl settings

**Comment by deeptim123** [comment]
There are currently no plans to support regression. @cartacioS  for visibility of this ask

**Comment by crisansou** [comment]
@deeptim123 ,

Thanks for the info. I think it's important to add this functionality for regression and classification as well.

> There are currently no plans to support regression. @cartacioS for visibility of this ask

**Comment by cartacioS** [comment]
@crisansou - this is currently not on our roadmap, and purposefully unprioritized as 90% of our customer base, especially customers investing in thousands+ models are leveraging only forecasting scenarios. Priorities change from one semester to the next, and it may be supported at a later date but is not in scope right now. If you are or have a direct customer who is blocked by the lack of many model support for regression and classification please contact me at sabina.cartacio@microsoft.com and we can further discuss.

Thanks!

**Comment by crisansou** [comment]
Hi @cartacioS ,

Got it, thanks for the info! The project is starting now, but if it is really necessary to use the multiple models solution for regression I'll send you an email.

> @crisansou - this is currently not on our roadmap, and purposefully unprioritized as 90% of our customer base, especially customers investing in thousands+ models are leveraging only forecasting scenarios. Priorities change from one semester to the next, and it may be supported at a later date but is not in scope right now. If you are or have a direct customer who is blocked by the lack of many model support for regression and classification please contact me at [sabina.cartacio@microsoft.com](mailto:sabina.cartacio@microsoft.com) and we can further discuss.
> 
> Thanks!

**Comment by CESARDELATORRE** [comment]
Closing as this is being tracked offline as a feature request for MANY MODELS, by Sabina.
